inspirehep/dojson/hep/fields/bd90x99x.py

```diff
@@ -24,36 +24,47 @@
 
 from dojson import utils
 
+from idutils import normalize_isbn
+
 from inspirehep.dojson import utils as inspire_dojson_utils
 from inspirehep.dojson.utils import strip_empty_values
 
 from ..model import hep, hep2marc
 
+from isbnlib._exceptions import NotValidISBNError
+
 
 @hep.over('references', '^999C5')
 def references(self, key, value):
     """Produce list of references."""
     value = utils.force_list(value)
 
     def get_value(value):
-        recid = None
-        number = ''
-        year = ''
-        if '0' in value:
-            try:
-                recid = int(value.get('0'))
-            except:
-                pass
-        if 'o' in value:
+        # FIXME: There might be multiple 0, o or y values. I.e. they might be
+        # tuples, not strings.
+        isbn = ''
+        try:
+            recid = int(value.get('0', ''))
+        except (TypeError, ValueError):
+            # FIXME: log.warning("Invalid value(s) in recid")
```

Comment:
We're planning to configure customized machine-friendly logger for errors inside dojson. Though It's not configured yet.